common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java

```diff
@@ -239,11 +382,93 @@ private void handleStateChange(LootrunningState oldState, LootrunningState newSt
             selectedBeacons.put(closestBeacon.color(), selectedBeacons.getOrDefault(closestBeacon.color(), 0) + 1);
             selectedBeaconsStorage.touched();
             WynntilsMod.postEvent(new LootrunBeaconSelectedEvent(closestBeacon));
+
+            // We selected a beacon, so other beacons are no longer relevant.
+            beacons.clear();
             setClosestBeacon(null);
+            LOOTRUN_BEACON_COMPASS_PROVIDER.reloadTaskMarkers();
+
             return;
         }
     }
 
+    private Pair<Double, TaskLocation> getBeaconTaskLocationPrediction(Beacon beacon) {
+        Optional<LootrunLocation> location = getLocation();
+        if (location.isEmpty()) {
+            WynntilsMod.warn("Location was when trying to predict for: " + beacon);
+            return null;
+        }
+
+        Set<TaskLocation> currentTaskLocations = taskLocations.get(location.get());
+        if (currentTaskLocations == null || currentTaskLocations.isEmpty()) {
+            WynntilsMod.warn("No task locations found for " + location.get());
+            return null;
+        }
+
+        List<TaskLocation> usedTaskLocations = beacons.entrySet().stream()
+                .filter(entry -> entry.getKey() != beacon.color())
+                .map(Map.Entry::getValue)
+                .map(Pair::b)
+                .toList();
+
+        double lowestPredictionScore = Double.MAX_VALUE;
+        TaskLocation closestTaskLocation = null;
+        for (TaskLocation currentTaskLocation : currentTaskLocations) {
+            // Possible optimization:
+            // Calculate the prediction score for tasks that are used by other beacons, and if our prediction score
+            // is lower than the other beacons, we can override it and calculate a new prediction for the other beacon.
+            if (usedTaskLocations.contains(currentTaskLocation)) {
+                continue;
+            }
```

Comment:
Actually.. this is not an option to implement. I will have to add this tomorrow otherwise we might predict correct locations but can't correctly assign it to colors.